docs/dev/software_development_process.md

```diff
@@ -0,0 +1,109 @@
+# USBNugget software development process
+
+## 1. Work planning and task creation
+
+For now, we will decide what work needs to be done during our weekly meeting.
+Work that needs doing is listed on [this project's Kanban
+board](https://github.com/orgs/HakCat-Tech/projects/4) under the "To do"
+column. Work that needs doing but is not fully defined can be added to the
+"Backlog" column.
+
+## 2. Communicate intended implementation
+
+Extremely trivial changes require no communication, other than updating the
+Kanban board to reflect what you'll be working on.
+
+When implementing non-trivial changes, share your ideas before implementing
+them.
+- For minor features, a short discussion during regular meeting or over chat is
+  probably sufficient
+- For major features, strongly consider creating a design document. This allows
+  other developers to comment on the design, suggest improvements or
+  alternatives, and think about edge cases. This prevents wasting time
+  constructing flawed implementations, and provides future developers with
+  artifacts to understand the code base and past decisions. Design documents
+  should be as short as possible while fully capturing your ideas. Use Google
+  docs or some other service that allows others to easily add comments in
+  specific sentences/paragraphs.
+
+## 3. Implementation
+
+### Style guide
```

Comment:
Idk where this fits but we might also add something for semantic versioning.  ie making sure the software version and all references in docs, images, etc match, to avoid confusion.  Reviewers should also look out for this.